app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -2229,9 +2231,14 @@ public void onDrmStateChanged(Session aSession, @SessionState.DrmState int aDrmS
 
     @Override
     public boolean onHandleExternalRequest(@NonNull String url) {
-        if (UrlUtils.isEngineSupportedScheme(url)) {
```

Comment:
The isEngineSupportedScheme is already parsing the url. Aren't we duplicating this task now ?